prov/util/src/util_ep.c

```diff
@@ -36,6 +36,38 @@
 #include <fi_enosys.h>
 #include <fi_util.h>
 
+int ofi_ep_bind_cq(struct util_ep *ep, struct util_cq *cq, uint64_t flags)
+{
+	if (flags & ~(FI_TRANSMIT | FI_RECV |
+		      FI_SELECTIVE_COMPLETION))
+		return -FI_EBADFLAGS;
+
+	if (((flags & FI_TRANSMIT) && ep->tx_cq) ||
+	    ((flags & FI_RECV) && ep->rx_cq))
+		return -FI_EINVAL;
+
+	if (flags & FI_TRANSMIT) {
+		ep->tx_cq = cq;
+		if (!(flags & FI_SELECTIVE_COMPLETION))
+			ep->tx_op_flags |= FI_COMPLETION;
```

Comment:
These flags have to be initialized in ofi_endpoint_init from the info arg. Same for rx_op_flags below.